src/coreclr/jit/morph.cpp

```diff
@@ -7754,6 +7754,16 @@ GenTree* Compiler::fgMorphCall(GenTreeCall* call)
 #endif
     }
 
+    if (call->gtCallMoreFlags & GTF_CALL_M_SPECIAL_INTRINSIC)
+    {
+        if (lookupNamedIntrinsic(call->AsCall()->gtCallMethHnd) ==
+            NI_System_Text_UTF8Encoding_UTF8EncodingSealed_GetUtf8Bytes)
+        {
+            // Expanded in fgVNBasedIntrinsicExpansion
+            setMethodHasSpecialIntrinsics();
+        }
```

Comment:
Why set this in morph instead of during import where we already have the named intrinsic available? The flag name also seems inaccurate, maybe rename it to `MethodHasExpandableSpecialIntrinsic`?

Reply:
I usually do this in morph because in importer we would need to transfer this flag from inlinee's compiler to the root one (or we can always set it in root's one but that is a bit conservative, we have a few cases where we reject inlining on the go, e.g. for generics with runtime lookups)